tests/phpunit/includes/testcase-multisites.php

```diff
@@ -0,0 +1,312 @@
+	 *
+	 * @param WP_Site $blog      Blog to set up.
+	 * @param array   $languages Languages to create.
+	 * @param array   $options   Polylang options to use.
+	 * @param string  $structure Permalink structure to use.
+	 * @return void
+	 */
+	protected function set_up_blog_with_pll( WP_Site $blog, array $languages, array $options, string $structure ) {
+		global $wp_rewrite;
+
+		switch_to_blog( $blog->blog_id );
+
+		$wp_rewrite->init();
+		$wp_rewrite->extra_rules_top = array();
+		$wp_rewrite->set_permalink_structure( $structure );
+
+		$plugins = get_option( 'active_plugins', array() );
+		update_option( 'active_plugins', array_merge( $plugins, $this->get_plugin_names() ) );
+
+		$pll_admin = $this->get_pll_admin_env( $options, false );
+
+		foreach ( $languages as $language ) {
+			$added = $pll_admin->model->add_language( $language );
+		}
+
+		$pll_admin->model->get_links_model()->init();
+		$pll_admin->init();
+		$wp_rewrite->flush_rules();
+
+		restore_current_blog();
+	}
+
+	/**
+	 * Sets a blog up with the correct structure and deactivates Polylang's plugins.
+	 *
+	 * @global $wp_rewrite
+	 *
+	 * @param WP_Site $blog      Blog to set up.
+	 * @param string  $structure Permalinks structure to use.
+	 * @return void
+	 */
+	protected function set_up_blog_without_pll( WP_Site $blog, string $structure ) {
+		global $wp_rewrite;
+
+		switch_to_blog( $blog->blog_id );
+
+		$wp_rewrite->init();
+		$wp_rewrite->extra_rules_top = array();
+		$wp_rewrite->set_permalink_structure( $structure );
+		$wp_rewrite->flush_rules();
+
+		$plugins = get_option( 'active_plugins', array() );
+		update_option( 'active_plugins', array_diff( $plugins, $this->get_plugin_names() ) ); // Ensure Polylang plugins are deactivated.
+
+		restore_current_blog();
+	}
+
+	/**
+	 * Removes *all* filters that could interfere with a test.
+	 * Must be called *before* creating a new Polylang environment in a test.
+	 */
+	protected function clean_up_filters() {
+		remove_all_filters( 'pll_init' );
+		remove_all_actions( 'pll_prepare_rewrite_rules' );
+		remove_all_actions( 'switch_blog' );
+
+		remove_all_filters( 'rewrite_rules_array' );
+		$types = array_merge( array( 'date', 'root', 'comments', 'search', 'author' ), get_post_types(), get_taxonomies() );
+
+		foreach ( $types as $type ) {
+			remove_all_filters( $type . '_rewrite_rules' );
+		}
+	}
+
+	/**
+	 * Returns Polylang's plugin basenames.
+	 *
+	 * @return string[]
+	 */
+	protected function get_plugin_names(): array {
+		return array( POLYLANG_BASENAME );
+	}
+
+	/**
+	 * Returns an instance of the main Polylang object along required instanciated classes for the tests.
+	 *
+	 * @param array $options Plugin options.
+	 * @param bool  $init    Trigger `PLL_Links_Model`'s and `PLL_Admin_Base`'s init or not. Default is `true`.
+	 * @return PLL_Admin_Base Polylang main class instance.
+	 */
+	protected function get_pll_admin_env( array $options = array(), bool $init = true ): PLL_Admin_Base {
```

Comment:
In fact, in always return `PLL_Admin`.

Reply:
Done in [6d6c6e7](https://github.com/polylang/polylang/pull/1352/commits/6d6c6e7c6ab110dbd623ff1a5685d597404d1bd7).